pkg/pillar/cmd/upgradeconverter/upgradeconverterutils.go

```diff
@@ -0,0 +1,50 @@
+package upgradeconverter
+
+import (
+	"encoding/json"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+// FilePathData - Stores old and new file paths
+type FilePathData struct {
```

Comment:
Why do we need this complexity?

The arguments to the convert command could be oldfile newfile.
And the delete etc only takes one file.
I can't see the intended usage since there is not yet any code which uses the converter so it is hard to tell the intent with this struct and associated json file(s).

Reply:
This is mainly for unit-testing. No hard-coded paths in upgrade converter. With this.. Except the parsing of CLI args, everything is unit-tested - including creation / modifying / deletion of files..